librosa/core/notation.py

```diff
@@ -471,6 +473,93 @@ def list_thaat() -> List[str]:
     """
     return list(THAAT_MAP.keys())
 
+@overload
+def __note_to_degree(key: str) -> int:
+    ...
+@overload
+def __note_to_degree(key: _IterableLike[str]) -> np.ndarray:
+    ...
+@overload
+def __note_to_degree(key: Union[str, _IterableLike[str], Iterable[str]]) -> Union[int, np.ndarray]:
+    ...
+def __note_to_degree(key: Union[str, _IterableLike[str], Iterable[str]]) -> Union[int,np.ndarray]:
+    """Take a note name and spit out the degree of that note (e.g. 'C#' -> 1). We allow possibilities like "C#b".
+
+    >>> librosa.__note_to_degree('B#')
+    0
+
+    >>> librosa.__note_to_degree('D♮##b')
+    3
+
+    >>> librosa.__note_to_degree(['B#','D♮##b'])
+    array([0,3])
+
+    """
+    if not isinstance(key, str):
+        return np.array([__note_to_degree(n) for n in key])
+
+
+    match = NOTE_RE.match(key)
+
+    if not match:
+        raise ParameterError(f"Improper key format: {key:s}")
+
+    letter = match.group('note').upper()
+    accidental = match.group('accidental')
+    pitch_map = {"C": 0, "D": 2, "E": 4, "F": 5, "G": 7, "A": 9, "B": 11}
+    counter = Counter(accidental)
+    return (pitch_map[letter]+sum([ACC_MAP[acc] * counter[acc] for acc in ACC_MAP]))%12
+
+@overload
+def __simplify_note(key: str, additional_acc: str =..., unicode: bool= ...) -> str:
+    ...
+
+@overload
+def __simplify_note(key: _IterableLike[str], additional_acc: str=..., unicode: bool = ... ) -> np.ndarray:
+    ...
+
+@overload
+def __simplify_note(key: Union[str, _IterableLike[str], Iterable[str]], additional_acc: str =..., unicode: bool = ...) -> Union[str, np.ndarray]:
+    ...
+
+def __simplify_note(key: Union[str, _IterableLike[str], Iterable[str]], additional_acc: str='', unicode: bool = True) -> Union[str, np.ndarray]:
+    """Take in a note name and simplify by canceling sharp-flat pairs, and doubling accidentals as appropriate.
+
+    >>> librosa.__simplify_note('C♭♯')
+    'C'
+
+    >>> librosa.__simplify_note('C♭♭♭')
+    'C♭𝄫'
+
+    >>> librosa.__simplify_note(['C♭♯', 'C♭♭♭'])
+    array(['C', 'C♭𝄫']
+
+    """
+    if not isinstance(key,str):
+        return np.array([__simplify_note(n+additional_acc, unicode=unicode) for n in key])
```

Comment:
Can you add a test case that covers this path, ie simplifying an array of keys?

Reply:
Sounds good. I'll get right on it!

By the way, is there a way I can do the lint and codecov checking locally so that I don't have to keep triggering the tests before I know whether they will pass?

Reply:
Yeah!  If you check the CONTRIBUTING guidelines, it should give you everything you need to set up a local test environment.  Then it should just be a matter of running the same commands that are in our github actions:
https://github.com/librosa/librosa/blob/09e4a622456bbad314e2e6d5c75879f56d728b9b/.github/workflows/lint_python.yml#L45-L71
and
https://github.com/librosa/librosa/blob/09e4a622456bbad314e2e6d5c75879f56d728b9b/.github/workflows/ci.yml#L106

It won't upload the coverage report to codecov, but you'll at least be able to see the coverage report locally.